File: python/src/bank/payments/emi.py
from __future__ import annotations

def emi(principal: float, annual_rate_pct: float, months: int) -> float:
    if months <= 0 or principal < 0:
        raise ValueError("Invalid inputs")
    r = annual_rate_pct / 12.0 / 100.0
    return (principal * r) / (1 - (1 + r) ** (-months)) if r != 0 else principal / months

File: javascript/src/payments/emi.js
export function emi(principal, annualRatePct, months) {
  if (months <= 0 || principal < 0) throw new Error("Invalid inputs");
  const r = annualRatePct / 12 / 100;
  return r !== 0 ? (principal * r) / (1 - (1 + r) ** (-months)) : principal / months;
}
